packages/itwinui-react/src/core/utils/components/Portal.tsx

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { ThemeContext } from '../../ThemeProvider/ThemeContext.js';
+import { getDocument } from '../functions/dom.js';
+import { useIsClient } from '../hooks/useIsClient.js';
+
+export type PortalProps = {
+  /**
+   * Where should the element be portaled to?
+   *
+   * If true, it will portal into nearest ThemeContext.portalContainer.
+   *
+   * If false, it will not be portaled.
+   *
+   * Otherwise, it will portal to the element passed to `to`.
+   */
+  portal?: boolean | { to: HTMLElement | (() => HTMLElement) };
```

Comment:
as mentioned in the description, the function helps avoid SSR issues by delaying when the value is the accessed. (`document` is invalid on the server, so it needs to be delayed to client)

`HTMLElement` is needed because that's the expected type for [`createPortal`'s second argument](https://react.dev/reference/react-dom/createPortal#parameters)